Performance&Fixes&new AI/Effects

dnqbob opened the pull request:
![1e](https://user-images.githubusercontent.com/13763394/94784727-29575600-0401-11eb-8c42-587360c59397.gif)
![newe1](https://user-images.githubusercontent.com/13763394/94784736-2bb9b000-0401-11eb-8a43-e0726214e491.gif)

dnqbob committed:
Optimazing & code simplify ion cannon 2

dnqbob committed:
New effect for coloss zaps

dnqbob committed:
Add new fire effect on YAML

dnqbob committed:
Make fire spawned by Tripod shorter

dnqbob committed:
New effect for SentryGun & code simplify fix

dnqbob committed:
Nolt's new jascgreenalpha.pal

dnqbob commented:
Punchbag AI will come in the next branch. We may need a discuss on it.

dnqbob committed:
Some spawned actors will give experience to killer

dnqbob commented:
![Uploading 1.gif…]()

ABrandau commented on mods/sp/rules/support.yaml:
```diff
@@ -1083,6 +1083,8 @@ MUMINE:
 		HP: 5000
 	Armor:
 		Type: Mine
+	GivesExperience:
```
Mines dont give experience iirc

dnqbob replied:
Will reduce it to 10

dnqbob replied:
fixed

dnqbob committed:
AI overfix and new AI

1. Add new Easy AI for just punchbag, old Easy AI change to Normal AI
- New Easy AI won't attack player, although it has the same build list on units and building.
- By doing this I can reuse some bot modules, which is very long and mess.
2. Minelayer of CABAL only be used for Cheater AI and Shatter AI
3. Fix YAML description of Iron Savior
4. Cash cheat adjust
4. Drop "HarvesterBotModule" because it is useless and laggy.

dnqbob commented:
New AI will use Normal AI (old Easy AI) build setting, but:
1.	It will not to use support power
2.	It will not attack player and have low protection desire, just build units and structure.

This will be enough for new players to deal with.
